rs/execution_environment/src/scheduler/round_schedule.rs

```diff
@@ -229,4 +231,77 @@ impl RoundSchedule {
 
         (canisters_partitioned_by_cores, canisters)
     }
+
+    /// Finishes canister execution on thread.
+    pub fn finish_canister_execution_on_thread(
+        canister: &mut CanisterState,
+        executed_priority_ids: &mut BTreeMap<CanisterId, AccumulatedPriority>,
+        round_id: ExecutionRound,
+        is_first_iteration: bool,
+        rank: usize,
+    ) {
+        let full_message_execution = match canister.next_execution() {
+            NextExecution::None => true,
+            NextExecution::StartNew => false,
+            // We just finished a full slice of executions.
+            NextExecution::ContinueLong | NextExecution::ContinueInstallCode => true,
```

Comment:
As commented on your earlier PR, I don't think a canister with `next_execution() == ContinueInstallCode` actually got anything done in an inner round so, unless I'm missing anything, we should not claim it got a full round.

Reply:
We executed install code previously, so IMO it's correct. It would be better to charge canisters right after the long install code. But we plan to move the install code in the round at some point, so I'd leave it as is...

Reply:
For the sake of time, I changed it.